### PR TITLE
Solution for Sponge Case Challenge

### PR DESCRIPTION
**Pull request by likechrisss**
**name:** solution review  
**about:** A template PR for code review with a checklist

---

### Description

SpongeCase is a style of text in which letters alternate between lowercase and uppercase, starting with lowercase. Each letter's case depends on its index (even or odd). Non-alphabetic characters remain in place but still affect the index count.

**Example:**

```python
spongecase("hello world")
# Output: "hElLo wOrLd"

spongecase("Python")
# Output: "pYtHoN"

spongecase("HELLO")
# Output: "hElLo"
```

---

### Files

- [x] The file name accurately describes the function’s behavior (e.g., `spongecase.py`).
- [x] There is a **module docstring** in the function file (`spongecase.py`).
- [x] The **test file’s name** matches the function file name, placed under `tests/test_spongecase.py`.
- [x] There is a **module docstring** in the test file.

---

### Unit Tests

- [x] The test class has a **helpful name** in PascalCase (e.g., `TestSpongeCase`).
- [x] The test class has a **docstring** explaining what it tests.
- [x] Each unit test has:
  - [x] A **helpful name**.
  - [x] A **clear docstring**.
  - [x] **Only one assertion**.
  - [x] **No extra logic**—just setup, exercise, and verify.
- [x] **All tests pass**.
- [x] There are tests for **defensive assertions** (if applicable).
- [x] There are tests for **boundary cases** (e.g., empty string, all caps, etc.).

---

### Function Docstring

- [x] The function’s behavior is **clearly described**.
- [x] The function’s **arguments** are described (type, purpose, assumptions).
- [x] The **return value** is described (type, assumptions).
- [x] Defensive assertions (if any) are documented using **Raises**.
- [x] Each assertion checks only one assumption.
- [x] Include **3 or more doctests** (or enough to illustrate usage and edge cases).

---

### The Function

- [x] The function’s **name** clearly describes its behavior.
- [x] The function’s **name** matches its file name.
- [x] The function has **correct type annotations** (e.g., `def spongecase(text: str) -> str:`).

---

### Strategy

**Do’s:**
- [x] Variable names help to understand the approach.
- [x] Comments are clear and describe the **strategy** (why) rather than the implementation (how).
- [x] Lines of code are spaced to highlight different stages of the strategy.

**Don’ts:**
- [x] The function’s strategy should not be **missing** from the documentation.
- [x] The code shouldn’t have more comments than logic. If so, consider simplifying the implementation.

---

### Implementation

- [x] The code passes **formatting checks** (e.g., Black, Ruff).
- [x] The code passes **Ruff linting** checks.
- [x] The code has no (reasonable) **Pylint** errors.
  - In code review, decide which Pylint errors are worth fixing.
- [x] Variable names use **snake_case** and are clear.
- [x] The code follows the chosen strategy as simply as possible.
- [x] The implementation is **as simple as possible** for that strategy.
- [x] The code includes any needed **defensive assertions** (with minimal logic).

---

Please review and check off each item above.  
Once all items are addressed, we can finalize this PR.  
Feel free to comment with any questions or clarifications.

**Thank you for reviewing the SpongeCase solution!**

### CODE REVIEW

**Review by RamonColmenares — COMMENTED** [review]
Hi!, I think is great to have the readme with the problem description but if you leave it there it can get a little confuse, I would recommend that you englobe both the solution and the readme on a folder, maybe like this:

```
└── 📁solutions
    └── 📁challenge_#
        └── __init__.py
        └── spongecase.py
        └── README.MD
    └── 📁tests
        └── 📁challenge_#
            └── __init__.py
            └── test_spongecase.py
```

also I saw other group that use their name as the folder name, that could be a good alternative:

```
└── 📁solutions
    └── 📁 chrismy_leprince
        └── __init__.py
        └── spongecase.py
        └── README.MD
    └── 📁tests
        └── 📁chrismy_leprince
            └── __init__.py
            └── test_spongecase.py
```

File: solutions/tests/test_spongecase.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        Mapping indices:
+            Index:  0 1 2 3 4  5 6   7   8  9
+            Char:   H e l l o  ? 1   #   A  b
+            Case:   h E l L o  ? 1   #   a  B
+            Result: "hElLo?1#aB"
+        """
+        self.assertEqual(spongecase("Hello?1#Ab"), "hElLo?1#aB")
+
+
+if __name__ == "__main__":
```
Comment (RamonColmenares) [inline solutions/tests/test_spongecase.py]: I don't think that line 81 and 82 are necessary, considering the way the tests are executed with the command:

`python -m unittest`

Comment (likechrisss) [reply solutions/tests/test_spongecase.py]: Thank you, i removed it.

**Review by RamonColmenares — APPROVED** [review]
LGTM

**Review by hectordacb — APPROVED** [review]
lgtm
